app/models/decision.rb

```diff
@@ -9,7 +9,22 @@ class Decision < ApplicationRecord
   validates :status, inclusion: { in: ["granted", "refused"],
     message: "Please select Yes or No" }
 
+  validate :ensure_correction, on: :update
+
   def comment_made?
     granted? && comment_met.present? || refused? && comment_unmet.present?
   end
+
+  private
+
+  def ensure_correction
+    return unless planning_application.awaiting_determination?
+    return unless user.reviewer?
```

Comment:
should this return on line 22 not be controlled by pundit? if a user isn't a reviewer are they allowed to access the decision model?

Reply:
We use the same decision model for both assessor and reviewer. You're right that Pundit controls their access to the UI where they would be able to make this change, but without the validation there would be nothing stopping a correction being applied to the assessor's decision if a REST call was made directly, so it's just another layer of security.